### ISSUE TITLE
Question on plugin usage for React.createContext

### ISSUE THREAD

**Issue by vict-shevchenko** [ArnaudBarre/eslint-plugin-react-refresh#73]
Hello, I am seeking some help to better understand how should I organise my components in order to minimize elslint error for  
`react-refresh/only-export-components` rule. 

### Case - When creating a React Context it is common practice do like this

```javascript
// code is simplified
const ThemeContext = React.createContext();

const ThemeProvider  = ({ theme, children }) => {
	return <ThemeContext.Provider value={theme}>{children}</ThemeContext.Provider>;
};

const useTheme = () => {
	return useContext(ThemeContext);
};

export { ThemeProvider, useTheme };  //Fast refresh only works when a file only exports components...
```
Is this correct, that I need to split this to 3 files for fast-refresh to work propertly? Or is this a false-positive and file can be kept as is, just some modification to plugin are required?

Thanks for your answer.

**Comment by ArnaudBarre** [comment]
Yes this is the most annoying change when you want to follow the rule.
If you don't do it here, any change to this file will create an HMR updates of all files and components using `useTheme` and `ThemeProvider`. While sometimes having HMR updates at the root component is fine, sometimes it can lead to some state loss when some component do a bit too much on useEffects.

Personally I use less and less contexts and more zustand store when the context is used as a singleton (one instance always available). Theme provider seems like a well suited usecase for this.

Also splitting in 2 is enough, `ThemeContext` & `useTheme` in one and `ThemeProvider` in the other

**Comment by vict-shevchenko** [comment]
Thank you a lot for clarifications.
